jl_renku/core.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2021 Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from git import Repo
+
+
+def renkulab_url_from_gitlab_url(gitlab_url):
```

Comment:
You might find this class useful - it's robust for all kinds of different ways that one can specify a repo URL: https://github.com/SwissDataScienceCenter/renku-python/blob/0cf51f64fe29be415980199e240fd04c3edfb604/renku/core/models/git.py#L82-L83

Reply:
You can see all the cases it covers here: https://github.com/SwissDataScienceCenter/renku-python/blob/a2e3297a318dc07c611d45eda059e73e67dd152a/tests/core/models/test_git.py#L27

Reply:
That is good to know. I think the implementation that is here should be thought of as a stub to get the interactions working. I think we would switch to using the renku-core-service before we release this. Or, we might drop the plug-in completely if https://github.com/SwissDataScienceCenter/renku/issues/1907 works out.